frontend/ui/widget/keyvaluepage.lua

```diff
@@ -435,31 +439,50 @@ function KeyValuePage:init()
                     self:goToPage(page)
                 end
             end,
+            ok_text = "Go to page",
         },
+        call_hold_input_on_tap = true,
         bordersize = 0,
-        margin = Screen:scaleBySize(20),
         text_font_face = "pgfont",
         text_font_bold = false,
     }
     self.page_info = HorizontalGroup:new{
-        self.page_return_arrow,
         self.page_info_first_chev,
         self.page_info_spacer,
         self.page_info_left_chev,
+        self.page_info_spacer,
         self.page_info_text,
+        self.page_info_spacer,
         self.page_info_right_chev,
         self.page_info_spacer,
         self.page_info_last_chev,
-        self.page_return_spacer,
     }
 
     local footer = BottomContainer:new{
         dimen = self.dimen:copy(),
         self.page_info,
     }
+    local page_return = BottomContainer:new{
+        dimen = self.dimen:copy(),
+        WidgetContainer:new{
+            dimen = Geom:new{
+                w = Screen:getWidth(),
+                h = self.page_return_arrow:getSize().h,
+            },
+            self.return_button,
+        }
+    }
 
     local padding = Size.padding.large
     self.item_width = self.dimen.w - 2 * padding
+
+    -- Account for the outer FrameContainer's padding'
+    footer.dimen.w = self.item_width
+    page_return.dimen.w = self.item_width
+    -- NOTE: Yep, padding * 1. I'm guessing BottomContainer never pads on one side or something?
+    footer.dimen.h = self.dimen.h - padding
+    page_return.dimen.h = self.dimen.h - padding
```

Comment:
Haven't checked what kind of things all these are.
But indeed, it seems BottomContainer centers by default:
`x = (self.dimen.x or 0) + math.floor((self.dimen.w - contentSize.w)/2),`

But this 1* padding feel strange.
I guess if you want the chevron+text centered, and this HorizontalSpan(padding)+page_return left aligned, you might need:
```
BottomContainer w = screen_width, containing:
  OverlapGroup, containing:
    LeftContainer or directly WidgetContainer(left_pad+page_return)
    CenterContainer w=screen_width containing chevrons + text
```

Reply:
On the upside, it matches menu to the pixel (as far as height is concerned, at least), so, eh ^^.